### ISSUE TITLE
[Bug]: source.include not work

### ISSUE THREAD

**Issue by ChenXplorer** [web-infra-dev/rsbuild#788]
### Version

```sh
System:
    OS: macOS 13.5
    CPU: (10) arm64 Apple M1 Pro
    Memory: 257.75 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 119.0.6045.199
    Safari: 16.6
  npmPackages:
    @rsbuild/core: ^0.1.5 => 0.1.5 
    @rsbuild/plugin-check-syntax: ^0.1.5 => 0.1.5 
    @rsbuild/plugin-react: ^0.1.5 => 0.1.5 
    @rsbuild/shared: ^0.1.5 => 0.1.5
```

### Details

import @rsbuild/plugin-check-syntax  and use source.include, but not work
<img width="1152" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/48044617/07ded1ee-0708-438b-928c-73bfe1d789f4">

### Reproduce link

NONE

### Reproduce Steps

1. yarn create rsbuild@latest
2. choose react + ts
3. yarn add @rsbuild/plugin-check-syntax query-string
4. config file
```ts
import { defineConfig } from '@rsbuild/core';
import { pluginCheckSyntax } from '@rsbuild/plugin-check-syntax';
export default defineConfig({
  tools:{
  },
  source: {
    include: [
      /\/node_modules\/query-string\//
    ],
  },
  plugins: [pluginCheckSyntax()],
});

```
5. App.tsx add code
```tsx
import queryString from 'query-string';

console.log(location.search);
//=> '?foo=bar'

const parsed = queryString.parse(location.search);
console.log(parsed);
//=> {foo: 'bar'}

console.log(location.hash);
//=> '#token=bada55cafe'

const parsedHash = queryString.parse(location.hash);
console.log(parsedHash);
//=> {token: 'bada55cafe'}

parsed.foo = 'unicorn';
parsed.ilike = 'pizza';

const stringified = queryString.stringify(parsed);
//=> 'foo=unicorn&ilike=pizza'

location.search = stringified;
// note that `location.search` automatically prepends a question mark
console.log(location.search);
```

6. yarn build

**Comment by chenjiahan** [comment]
What is your target ECMAScript version?

- If the project targets to ES6 / ES5, you can adjust browserslist config, ref: https://rsbuild.dev/guide/advanced/browserslist

- If the project targets to ES2016+, you need to set the `ecmaVersion` version of plugin-check-syntax, ref: https://rsbuild.dev/plugins/list/plugin-check-syntax#ecmaversion

**Comment by chenjiahan** [comment]
If the project targets to ES5, you also needs to include the sub-dependencies of `query-string`.

```ts
export default defineConfig({
  source: {
    include: [
      /\/node_modules\/query-string\//,
      /\/node_modules\/decode-uri-component\//,
      /\/node_modules\/filter-obj\//,
      /\/node_modules\/split-on-first\//,
    ],
  },
  plugins: [pluginCheckSyntax()],
});
```

Ref: https://rsbuild.dev/config/options/source#compile-sub-dependencies

**Comment by ChenXplorer** [comment]
thx，it works, 
but i have a question, 
<img width="550" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/48044617/c5f53147-541f-406a-b165-30824d2442e4">
how can i  solve  these?

**Comment by chenjiahan** [comment]
No idea.. can you provide a reproduce repo for this?

**Comment by ChenXplorer** [comment]
1. add a file  , like:  a.module.css 
  ```css
  .xxx{
      height: 300;
  }
  ```
2. App.tsx add code 
```
import styles from './a.module.css';
 <div className={styles.xxx}>
 </div>
```
3. yarn build
4. Search in  dist ，u can find  code  like this
```
e.exports={xxx:"xxx-_0b695",xxx:"xxx-_0b695"}}
```
5. this is Redefinition of property problem.

**Comment by chenjiahan** [comment]
I have reproduced the issue, and I will continue to locate the cause of the problem

**Comment by chenjiahan** [comment]
This is a bug of Rspack, and I have created an issue in the Rspack repo: https://github.com/web-infra-dev/rspack/issues/4844

**Comment by chenjiahan** [comment]
Fixed in 

- https://github.com/web-infra-dev/rspack/issues/4844
- https://github.com/web-infra-dev/rsbuild/pull/860

Will be released in v0.2.0 (this week)
